### PR TITLE
Migrate issue 240

### LINKED ISSUE #240: Migrate "Templates with identical layouts" module to V4 report
### Overview
We should migrate the [Templates with identical layouts](https://github.com/Kentico/KInspector/blob/master/KInspector.Modules/Modules/Content/IdenticalTemplateLayoutsModule.cs) module to the new V4 report format.

### Definition of done
- [ ] Code and Scripts are ported to `KenticoInspector.Reports` project in a dedicated folder for the report.
- [ ] Scripts are referenced via constants in a static class.
- [ ] Scripts are refactored to return simple results that are mappable to simple classes.
- [ ] Report logic is covered by unit test for a clean result
- [ ] Report logic is covered by unit tests for all known dirty results
- [ ] Useful, non-specific logic is abstracted to services or helpers.

_Note: The [Class/Table Validation report](https://github.com/Kentico/KInspector/tree/v4-dev/KenticoInspector.Reports/ClassTableValidation) (and it's [tests](https://github.com/Kentico/KInspector/blob/v4-dev/KenticoInspector.Reports.Tests/ClassTableValidationTests.cs)) is a good, simple example of the main concepts._

### Additional Details
- New name: __Template Layout Analysis__
- Tags: __Portal Engine, Information__

### PR DESCRIPTION
**Pull request by kentico-anthonym**
### Motivation

Which issue does this fix? Fixes #240 

If no issue exists, what is the fix or new feature? Were there any reasons to fix/implement things that are not obvious?

### Checklist

- [x] Code follows coding conventions held in this repo
- [x] Automated tests have been added
- [x] Tests are passing
- [ ] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### CODE REVIEW

File: KenticoInspector.Reports.Tests/TemplateLayoutAnalysisTest.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                .Setup(p => p.ExecuteSqlFromFile<IdenticalPageLayouts>(Scripts.GetIdenticalLayouts))
+                .Returns(identicalPageLayouts);
+            // Act
+            var results = _mockReport.GetResults(_mockInstance.Guid);
+            // Assert
+            Assert.That(results.Data.Rows.Count != 0);
+            Assert.That(results.Status == ReportResultsStatus.Information);
+        }
+
+        public void Should_ReturnEmptyListOfIdenticalLayouts()
```
Comment (ChristopherJennings) [inline KenticoInspector.Reports.Tests/TemplateLayoutAnalysisTest.cs]: Missing the [Test] attribute. Also, this should include a "when" condition in the name. E.g. Should_ReturnEmptyListOfIdenticalLayouts_WhenNoneFound

File: KenticoInspector.Reports.Tests/TemplateLayoutAnalysisTest.cs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        private Report _mockReport;
+
+        public TemplateLayoutAnalysisTests(int majorVersion)
+        {
+            InitializeCommonMocks(majorVersion);
+            _mockReport = new Report(_mockDatabaseService.Object, _mockInstanceService.Object);
+        }
+
+        [Test]
+        public void Should_ReturnListOfIdenticalLayouts()
```
Comment (ChristopherJennings) [inline KenticoInspector.Reports.Tests/TemplateLayoutAnalysisTest.cs]: This should include a "when" condition in the name. E.g. Should_ReturnEmptyListOfIdenticalLayouts_WhenSomeAreFound

File: KenticoInspector.Reports.Tests/TemplateLayoutAnalysisTest.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        {
+            // Arrange
+            var identicalPageLayouts = GetListOfLayouts();
+            _mockDatabaseService
+                .Setup(p => p.ExecuteSqlFromFile<IdenticalPageLayouts>(Scripts.GetIdenticalLayouts))
+                .Returns(identicalPageLayouts);
+            // Act
+            var results = _mockReport.GetResults(_mockInstance.Guid);
+            // Assert
+            Assert.That(results.Data.Rows.Count != 0);
```
Comment (ChristopherJennings) [inline KenticoInspector.Reports.Tests/TemplateLayoutAnalysisTest.cs]: This should be a more specific assertion. Given the test conditions, we should know that exactly 5 sets of identical layouts are returned, so we should check for that. For example, if only 4 are returned after processing, the test would still pass, but the logic in the report is wrong.

**Comment by kentico-anthonym** [comment]
All review issues have been resolved.

**Pull request merged**